src/test/java/edu/cornell/kfs/krad/service/impl/CuAttachmentServiceImplTest.java

```diff
@@ -47,28 +50,110 @@
+        attachmentService = null;
+    }
+    
+    private NoteService buildMockNoteService() {
+        NoteService mockNoteService = Mockito.mock(NoteService.class);
+        Mockito.when(mockNoteService.getNoteByNoteId(Mockito.any())).thenReturn(noteVirus);
+        return mockNoteService;
+    }
+    
+    private Note setupMockNote(String remoteObjectIdentifier) {
+        Note mockNote = Mockito.mock(Note.class);
+        Mockito.when(mockNote.getRemoteObjectIdentifier()).thenReturn(remoteObjectIdentifier);
+        return mockNote;
+    }
+    
+    private CuAttachmentDaoOjb buildMockAttachmentDao() {
+        CuAttachmentDaoOjb mockDao = Mockito.mock(CuAttachmentDaoOjb.class);
+        Mockito.when(mockDao.getAttachmentByAttachmentId(Mockito.any())).then(this::buildAttachment);
+        return mockDao;
+    }
+    
+    private Attachment buildAttachment(InvocationOnMock invocation) {
+        String attachmentIdentifier = invocation.getArgument(0);
+        Attachment attachment = new Attachment();
+        attachment.setObjectId(attachmentIdentifier);
+        attachment.setAttachmentIdentifier(attachmentIdentifier);
+        Note note = setupMockNote(attachmentIdentifier);
+        attachment.setNote(note);
+        return attachment;
+    }
+    
+    private ConfigurationService buildMockConfigurationService() {
+        ConfigurationService mockConfigurationService = Mockito.mock(ConfigurationService.class);
+        Mockito.when(mockConfigurationService.getPropertyValueAsString(Mockito.any())).then(this::getConfigurationPropertyAsString);
+        return mockConfigurationService;
+    }
+    
+    private String getConfigurationPropertyAsString(InvocationOnMock invocation) {
+        String inputString = invocation.getArgument(0);
+        if (StringUtils.equals(inputString, KRADConstants.ATTACHMENTS_DIRECTORY_KEY)) {
+            return TEST_ATTACHMENTS_PATH;
+        }
+        return null;
+    }
+    
+    private AntiVirusService buildMockAntiVirusService() {
+        AntiVirusService mockAntivirusService = Mockito.mock(AntiVirusService.class);
+        Mockito.when(mockAntivirusService.scan(Mockito.any(InputStream.class))).then(this::getScanResultFromInputStream);
+        return mockAntivirusService;
+    }
+    
+    private ScanResult getScanResultFromInputStream(InvocationOnMock invocation) {
+        InputStream scannedInputStream = invocation.getArgument(0);
+        String scannedContents;
+        try {
+            scannedContents = IOUtils.toString(scannedInputStream);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        ScanResult result = Mockito.mock(ScanResult.class);
+        
+        if (StringUtils.equals(scannedContents, GOOD_FILE_CONTENTS)) {
+            Mockito.when(result.getStatus()).thenReturn(ScanResult.Status.PASSED);
+        } else {
+            Mockito.when(result.getStatus()).thenReturn(ScanResult.Status.FAILED);
+        }
```

Comment:
Scan results also have an ERROR status. Would it be worth testing that in this test class as well?